ranker/modules/omnicorp_overlay.py

```diff
@@ -21,99 +20,67 @@
 CACHE_PASSWORD = os.environ.get("CACHE_PASSWORD", "")
 
 
-async def count_node_pmids(supporter, node, key, value, cache, kgraph):
-    """Count node PMIDs and add as node property."""
-    if value is not None:
-        logger.debug(f"{key} is cached")
-        support_dict = value
-    else:
-        logger.debug(f"Computing {key}...")
-        support_dict = await supporter.node_pmid_count(node)
-        if cache and support_dict["omnicorp_article_count"]:
-            cache.set(key, support_dict)
-
-    # add omnicorp_article_count to nodes in networkx graph
-    attribute = {
-        "original_attribute_name": "omnicorp_article_count",
-        "attribute_type_id": "biolink:has_count",
-        "value": support_dict["omnicorp_article_count"],
-        "value_type_id": "EDAM:data_0006",
-    }
-
-    # if there is no attributes array add one.  We used exclude_None=True so we don't need to check for it here.
-    if "attributes" not in kgraph[node]:
-        kgraph[node]["attributes"] = []
-
-    # save the attributes
-    kgraph[node]["attributes"].append(attribute)
-
-
-async def count_shared_pmids(
-    supporter,
-    support_idx,
-    pair,
-    key,
-    value,
-    cache,
-    cached_prefixes,
+async def add_node_pmid_counts(kgraph, counts):
+    for node_id in kgraph["nodes"]:
+        if node_id in counts:
+            count = counts[node_id]
+        else:
+            count = 0
```

Comment:
Do we want to add this node attribute if the count is 0? Or just leave it out?

Reply:
After more thought, this makes sense. It would actually be confusing if some nodes didn't have this attribute after calling the overlay service. So this is good.